src/app/models/harvest/Harvest.ts

```diff
@@ -0,0 +1,98 @@
+import { Injector } from "@angular/core";
+import { projectHarvestRoute } from "@components/projects/projects.routes";
+import {
+  DateTimeTimezone,
+  HasCreatorAndUpdater,
+  Id,
+} from "@interfaces/apiInterfaces";
+import { AbstractModelWithoutId } from "../AbstractModel";
+
+/**
+ * Status of a harvest
+ *
+ * State transition map:
+ * ```
+ *                     |-------------------------------(streaming only)-------------------|
+ *                     ↑                                                                  ↓
+ * :new_harvest → :uploading → :metadata_extraction → :metadata_review → :processing → :complete
+ *                     ↑                ↑                      ↓
+ *                     |----------(batch upload only)----------|
+ *```
```

Comment:
be worth linking to the baw-server file where this state machine is defined. The link will probably be broken until the relevant changes are merged into master for baw-server, but regardless...